Makefile

```diff
@@ -15,111 +15,96 @@
+# Application allowed derivation curves.
+CURVE_APP_LOAD_PARAMS = ed25519
 
-## USB HID?
-DEFINES += HAVE_IO_USB HAVE_L4_USBLIB IO_USB_MAX_ENDPOINTS=4 IO_HID_EP_LENGTH=64 HAVE_USB_APDU
+# Application allowed derivation paths.
+PATH_APP_LOAD_PARAMS += "44'/1815'" "1852'/1815'" "1853'/1815'" "1854'/1815'" "1855'/1815'" "1694'/1815'"
 
-## USB U2F
-DEFINES += HAVE_U2F HAVE_IO_U2F U2F_PROXY_MAGIC=\"ADA\" USB_SEGMENT_SIZE=64
+# Setting to allow building variant applications
+VARIANT_PARAM = COIN
+VARIANT_VALUES = cardano_ada
 
-## WEBUSB
-#WEBUSB_URL = https://www.ledger.com/pages/supported-crypto-assets
-#DEFINES += HAVE_WEBUSB WEBUSB_URL_SIZE_B=$(shell echo -n $(WEBUSB_URL) | wc -c) WEBUSB_URL=$(shell echo -n $(WEBUSB_URL) | sed -e "s/./\\\'\0\\\',/g")
-DEFINES   += HAVE_WEBUSB WEBUSB_URL_SIZE_B=0 WEBUSB_URL=""
+DEFINES += RESET_ON_CRASH
+# Enabling DEBUG flag will enable PRINTF and disable optimizations
+# DEVEL = 1
+# DEFINES += HEADLESS
 
-## BLUETOOTH
-ifeq ($(TARGET_NAME),$(filter $(TARGET_NAME),TARGET_NANOX TARGET_STAX))
-	DEFINES += HAVE_BLE BLE_COMMAND_TIMEOUT_MS=2000 HAVE_BLE_APDU
+# Enabling debug PRINTF
+ifeq ($(DEVEL), 1)
+	DEBUG = 1
+	DEFINES += DEVEL
 endif
 
-## Protect stack overflows
-DEFINES += HAVE_BOLOS_APP_STACK_CANARY
-
+########################################
+#     Application custom permissions   #
+########################################
+HAVE_APPLICATION_FLAG_BOLOS_SETTINGS = 1
+
+########################################
+#         NBGL custom features         #
+########################################
+ENABLE_NBGL_QRCODE = 1
+
+########################################
+# Application communication interfaces #
+########################################
+ENABLE_BLUETOOTH = 1
+
+########################################
+#          Features disablers          #
+########################################
+# These advanced settings allow to disable some feature that are by
+# default enabled in the SDK `Makefile.standard_app`.
+DISABLE_STANDARD_APP_FILES = 1
 ifeq ($(TARGET_NAME),TARGET_NANOS)
-DEFINES += IO_SEPROXYHAL_BUFFER_SIZE_B=128
-else
-DEFINES += IO_SEPROXYHAL_BUFFER_SIZE_B=300
-DEFINES += HAVE_GLO096
-ifneq ($(TARGET_NAME),TARGET_STAX)
-DEFINES += HAVE_BAGL BAGL_WIDTH=128 BAGL_HEIGHT=64
-DEFINES += HAVE_UX_FLOW
+DISABLE_STANDARD_BAGL_UX_FLOW = 1
 endif
 
-ifeq ($(TARGET_NAME),TARGET_STAX)
-DEFINES += NBGL_QRCODE
-SDK_SOURCE_PATH += qrcode
-endif
-DEFINES += HAVE_BAGL_ELLIPSIS # long label truncation feature
-DEFINES += HAVE_BAGL_FONT_OPEN_SANS_REGULAR_11PX
-DEFINES += HAVE_BAGL_FONT_OPEN_SANS_EXTRABOLD_11PX
-DEFINES += HAVE_BAGL_FONT_OPEN_SANS_LIGHT_16PX
-endif
+SDK_SOURCE_PATH += lib_u2f
 
-DEFINES += RESET_ON_CRASH
+##############
+#  Compiler  #
+##############
+
+# based in part on https://interrupt.memfault.com/blog/best-and-worst-gcc-clang-compiler-flags
+WERROR   := -Werror=return-type -Werror=parentheses -Werror=format-security
```

Comment:
Flags and errors that are not in the standard app, not sure we should keep them :man_shrugging: @tdejoigny-ledger @fbeutin-ledger

Reply:
Done.